service/worker/workerdeployment/client.go

```diff
@@ -357,27 +298,27 @@ func (d *ClientImpl) SetCurrentVersion(
 		requestID,
 	)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	if failure := outcome.GetFailure(); failure.GetApplicationFailureInfo().GetType() == errNoChangeType {
-		return nil, nil, serviceerror.NewAlreadyExist(fmt.Sprintf("Build ID %q is already current for %q",
+		return nil, serviceerror.NewAlreadyExist(fmt.Sprintf("Version %q is already current for %q",
```

Comment:
I think you should just return nil error here, and response with `PreviousVersion` = whatever the customer just tried to make current.